main.cc

```diff
@@ -325,77 +298,76 @@ main(int argc, char* argv[])
   waypt_init();
   route_init();
 
-  if (argc < 2) {
-    usage(argv[0],1);
+  if (qargs.size() < 2) {
+    usage(prog_name,1);
     exit(0);
   }
 
   /*
    * Open-code getopts since POSIX-impaired OSes don't have one.
    */
   argn = 1;
-  while (argn < argc) {
-    char* optarg;
+  while (argn < qargs.size()) {
+    QString optarg;
 
-    if (argv[argn][0] != '-') {
+//  we must check the length for afl input fuzzing to work.
+//    if (qargs.at(argn)[0].toLatin1() != '-') {
+    if (qargs.at(argn).size() > 0 && qargs.at(argn)[0].toLatin1() != '-') {
```

Comment:
1. "The QString::at() function can be faster than operator[](), because it never causes a deep copy to occur."  So it is probably better to use at for read only access as we are doing here, and more than just a choice of style.
2. qargs.at(argn).at(0) is a QChar, so we could do qargs.at(argn).at(0) != QChar('-'), but I think that is less efficient as the QChar construction will involve a QTextCodec.  QChar::toLatin1 in an inline define to return an existing private field.

Reply:
In 1, should have read "... faster than the \[\] operator ..."